mitodl/mit-open
# add UserList modals and wire up buttons

### What are the relevant tickets?
Closes https://github.com/mitodl/mit-open/issues/715

### Description (What does it do?)
This PR adds the appropriate modals for creating, editing and deleting a `UserList`. The modals were wired up to buttons in the appropriate places:

 - "Create new list" button in `UserListListingPage`
 - A context menu was added to the cards displayed on `UserListListingPage` containing edit and delete buttons
 - Edit button on `UserListDetailsPage`

### Screenshots (if appropriate):
<img width="959" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/fe32f186-0768-419b-b007-0dd2dc02e1ee">
<img width="959" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/3ecb3dd7-4afd-4044-9305-52f04861b0dc">
<img width="959" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/b78a2557-5540-4a58-8e90-095312e65498">
<img width="959" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/60a867f1-5737-4b33-8702-76eba8404e6a">
<img width="959" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/f4fe74d7-8bd0-495b-8c64-80d061224f4d">

<img width="146" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/9975af37-d706-4738-b990-e225dc417287">
<img width="147" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/27953106-206a-417d-8481-c35fb4bd1da8">
<img width="148" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/c2f093a3-6867-45e6-9d2e-05b42088d625">
<img width="147" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/f4fa56fc-2bcd-4ebe-a393-23dfca2e4b6c">

### How can this be tested?
 - Spin up `mit-open` on this branch
 - Visit http://localhost:8063/userlists/
 - Click the "Create new list" button to create a new list
 - Fill out the form and hit save
 - Verify that your new User List shows up in the listing page
 - Click the title of your new User List to enter the details page
 - Verify that the title and description are appropriately rendered
 - Click the Edit button in the upper right
 - Change the title and description and hit save
 - Verify that your changes are reflected on the details page
 - Navigate back to the listing page and verify once again that the changes to the title and description are reflected
 - Click the 3 dots on the card for your new User List and click the Edit button
 - Verify once again that editing the title and description reflects your changes on both the listing and details pages
 - Click the 3 dots again and click Delete, then click "Yes, Delete"
 - Verify that your User List is now gone from the listing page
